org-ql.el

```diff
@@ -448,6 +454,41 @@ Returns cons (INHERITED-TAGS . LOCAL-TAGS)."
                  org-ql-tags-cache))
       (puthash position all-tags tags-cache))))
 
+(defun org-ql--expand-tag-hierarchy (tags &optional excluded)
+  "Return TAGS along with their associated group tags.
+This function recursively searches for groups that each given tag belongs to,
+directly or indirectly, and includes the corresponding group tags to the result.
+
+TAGS should be a list of tags (i.e., strings).
+If non-nil, EXCLUDED should be a list of group tags that will not be
+automatically added to the results unless they are already in TAGS."
+  (let ((groups (org-tag-alist-to-groups org-current-tag-alist))
+        (excluded (append tags excluded))
+        result)
```

Comment:
Byte-compiling this function warns that `result` is unused.

Reply:
My bad. Leftovers from an initial draft of `org-ql--expand-tag-hierarchy`. It will be removed in the new commit.